src/pykeen/models/baseline/models.py

```diff
@@ -138,3 +140,54 @@ def score_h(self, rt_batch: torch.LongTensor, slice_size: Optional[int] = None)
             per_relation=self.head_per_relation,
             num_entities=self.num_entities,
         )
+
+
+class SoftInverseTripleBaseline(EvaluationOnlyModel):
+    """Score based on relation similarity.
+
+    ---
+    name: Soft Inverse Triple Baseline
+    citation:
+        author: Berrendorf
+        year: 2021
+        link: https://github.com/pykeen/pykeen/pull/543
+        github: pykeen/pykeen
+    """
+
+    def __init__(
+        self,
+        triples_factory: CoreTriplesFactory,
+        threshold: Optional[float] = None,
+    ):
+        super().__init__(triples_factory=triples_factory)
+        # compute relation similarity matrix
+        self.sim = get_relation_similarity(triples_factory, to_inverse=False, threshold=threshold)
+        self.sim_inv = get_relation_similarity(triples_factory, to_inverse=True, threshold=threshold)
+
+        mapped_triples = numpy.asarray(triples_factory.mapped_triples)
+        self.rel_to_head = scipy.sparse.coo_matrix(
```

Comment:
use `entity_pair_matrix()`?

Reply:
No, this is a relation-to-entity matrix, rather than a relation-to-entity-pairs matrix (the latter shape `num_entities**2` columns)

Reply:
worth making an external function since this code is duplicated? would be another chance to add documentation :)

Reply:
[0e658d3](https://github.com/pykeen/pykeen/pull/543/commits/0e658d394f4ea9eb5b93e0268b64a606b9492852)